src/Hardware.ino

```diff
@@ -7,13 +7,15 @@
 void hardwareInit()
 {
   // set GPIO pins state if not set to default
+  constexpr byte maxStates = sizeof(Settings.PinBootStates)/sizeof(Settings.PinBootStates[0]);
```

Comment:
Took me a while before I realized you just check for the nr of boot state definitions in the settings.
Maybe `nr_pinbootstate_settings` is more descriptive?
Or perhaps you can think of a better name here?